packages/gatsby-source-sensenet/src/utils.ts

```diff
@@ -0,0 +1,52 @@
+import { defaultRepositoryConfiguration, ODataUrlBuilder } from '@sensenet/client-core'
+import { SourceNodesArgs } from 'gatsby'
+import fetch from 'node-fetch'
+import { PluginConfig } from './gatsby-node'
+
+export const snPrefix = 'sensenet'
+
+export const createTreeNode = async (
+  parentNode: any,
+  content: any,
+  sourceNodesArgs: Partial<SourceNodesArgs>,
+  options: PluginConfig,
+  token: string,
+) => {
+  const newNode = {
+    ...content,
+    id: sourceNodesArgs.createNodeId!(`${content.Type}-${content.Id}`),
+    internal: {
+      type: `${snPrefix}${content.Type}`,
+      contentDigest: sourceNodesArgs.createContentDigest!(content),
+      description: `${content.Type} node`,
+    },
+  }
+
+  sourceNodesArgs.actions!.createNode(newNode)
+  sourceNodesArgs.actions!.createParentChildLink({ parent: parentNode, child: newNode })
+
+  try {
+    if ((options.level && options.level > 0) || !options.level) {
+      const params = ODataUrlBuilder.buildUrlParamString(defaultRepositoryConfiguration, options.oDataOptions)
+      console.info(
+        'REQUEST SENT TO SENENET:',
+        `${options.host}/${defaultRepositoryConfiguration.oDataToken}${content.Path}?${params}`,
+      )
+      const res = await fetch(`${options.host}/${defaultRepositoryConfiguration.oDataToken}${content.Path}?${params}`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        method: 'GET',
+      })
```

Comment:
It is working fine, but you could have also created a Repository instance, and use its load/loadCollection functions.

Reply:
I started with your suggestion first, but it is not so easy as it sounds. Every load and loadChildren has to know the type of the contents in the response. If you have a custom type how can you define it? It should work for every type. And "default content types" don't contain e.g.: Blog and BlogPost types. So I faced so many problems.